scantool/scantool.c

```diff
@@ -640,6 +640,15 @@ l2_do_send(struct diag_l2_conn *d_conn, void *data, size_t len, void *handle)
 
 	msg.len = len;
 	msg.data = (uint8_t *)data;
+
+	if (d_conn->l2proto->diag_l2_protocol == DIAG_L2_PROT_VAG) {
```

Comment:
what I mean is, why not set msg.type from the 850 sendreq ? What I'm trying to avoid is giving meaning to the first byte in data[] like this, for only one protocol, and from within a common code path.
I'm not sure what you mean by "other L2s that use the first byte of the message" ? I think only the l2_send for the MB1 protocol does anything special to data[0] ?